encoding/basics-and-workflow.md

```diff
@@ -29,7 +29,144 @@ provided by VSEdit.
+and just encode the file from the beginning to the end.
+
+To encode audio piped from ffmpeg,
+replace the input filename with a hyphen
+and place the whole command after the ffmpeg command,
+like this:
+```
+ffmpeg -i "input.dts" -acodec pcm_s24le -f wav -| flac -8 --ignore-chunk-sizes - -o "output.flac"
+```
+
+
+### Encoding with qaac
+
+First, set up qaac:
+* go to [qaac's download page][qaac] and download the newest build
+(at the time of writing, `qaac_2.68.zip`) and the `makeportable.zip` file
+* extract the `x64` folder wherever you want your qaac folder to be,
+then extract contents of `makeportable.zip` to it
+* download the [iTunes installation file][itunes] (`iTunes64Setup.exe`)
+and move it to the `x64` folder.
+* run the `makeportable.cmd` script
+* done, you can now delete the iTunes installation file.
+
+To encode from a file, use the following command:
+```
+qaac64 --tvbr 91 --ignorelength --no-delay "input.wav" -o "output.m4a"
+```
+The `--tvbr 91` option sets the encoding mode to True Variable Bitrate
+(in other words, constant quality)
+and sets the desired quality.
+Qaac has only 15 actual quality steps in intervals of 9 (0, 9, 18 ... 127).
+The higher the number, the higher the resulting bitrate will be.
+The recommended value is 91, which will result in bitrates
+of about 192 kbps, enough for complete transparency
+in vast majority of cases.  
+`--ignorelength` performs the same function as
+`--ignore-chunk-sizes` in FLAC.  
+`--no-delay` is needed for proper audio/video sync.
+Read why in this [HydrogenAudio forum post][hydrogen_forum].
+
+To encode audio piped from ffmpeg,
+replace the input filename with a hyphen
+and place the whole command after the ffmpeg command:
+```
+ffmpeg -i "input.dts" -acodec pcm_s24le -f wav - | qaac64 --tvbr 91 --ignorelength --no-delay  - -o "output.m4a"
+```
+
+[hydrogen_forum]: https://hydrogenaud.io/index.php/topic,85135.msg921707.html#msg921707
+[itunes]: https://secure-appldnld.apple.com/itunes12/031-69284-20160802-7E7B2D20-552B-11E6-B2B9-696CECD541CE/iTunes64Setup.exe
+[qaac]: https://sites.google.com/site/qaacpage/cabinet
+
+
+### Encoding with opusenc
+
+Encoding from a file with opusenc will look like this:
+```
+opusenc --bitrate 160 --vbr --ignorelength "input.wav" "output.opus"
+```
+As you may have noticed,
+opusenc CLI uses bitrate control,
+insted of some kind constant quality mode, like in qaac.
+This doesn't mean it works in constant bitrate mode—[the target bitrate is calibrated against the internal constant quality targets
+so that over a typical music collection,
+something very close to the target bitrate will be achieved][opus_hydrogen].
+This makes choosing the proper target quality
+a little bit funky—you will have to set different bitrate for stereo and for multichannel files.
+As per HydrogenAudio recommendation,
+the recommended bitrate for stereo is 160 kbps,
+while for 5.1 something in the neighbourhood of 320-384 kbps should be OK.  
```

Comment:
```suggestion
while for 5.1 something in the range of 320-384 kbps should be OK.  
```

If the two spaces at the end of the line are intended, use `<br>` to add a line break instead. Those are resistent to removal of trailing whitespace.